lib/widgets/emoji_reaction.dart

```diff
@@ -0,0 +1,264 @@
+              //
+              // Separately, web has 1px less than this on the left, but that
+              // asymmetry doesn't seem to help us.
+              padding: const EdgeInsets.symmetric(horizontal: 5, vertical: 2),
+              child: Row(mainAxisSize: MainAxisSize.min, crossAxisAlignment: CrossAxisAlignment.center, children: [
+                Padding(padding: const EdgeInsets.symmetric(horizontal: 3, vertical: 1),
+                  child: (() {
+                    if (emojiset == Emojiset.text) {
+                      return _TextEmoji(emojiName: emojiName, selected: selfVoted);
+                    }
+                    switch (reactionType) {
+                      case ReactionType.unicodeEmoji:
+                        return _UnicodeEmoji(
+                          emojiCode: emojiCode,
+                          emojiName: emojiName,
+                          selected: selfVoted,
+                        );
+                      case ReactionType.realmEmoji:
+                      case ReactionType.zulipExtraEmoji:
+                        return _ImageEmoji(
+                          emojiCode: emojiCode,
+                          emojiName: emojiName,
+                          selected: selfVoted,
+                        );
+                    }
+                  })()),
+                Flexible(
+                  // Added vertical: 1 to give some space when the label is
+                  // taller than the emoji (e.g. because it needs multiple lines)
+                  child: Padding(padding: const EdgeInsets.symmetric(horizontal: 3, vertical: 1),
+                    child: Text(
+                      textWidthBasis: TextWidthBasis.longestLine,
+                      style: TextStyle(
+                        fontFamily: 'Source Sans 3',
+                        fontSize: (14 * 0.90),
+                        height: 13 / (14 * 0.90),
+                        color: labelColor,
+                      ).merge(selfVoted
+                        ? weightVariableTextStyle(context, wght: 600, wghtIfPlatformRequestsBold: 900)
+                        : weightVariableTextStyle(context)),
+                      label)),
+                ),
+              ])))));
+  }
+}
+
+class _UnicodeEmoji extends StatelessWidget {
+  const _UnicodeEmoji({
+    required this.emojiCode,
+    required this.emojiName,
+    required this.selected,
+  });
+
+  final String emojiCode;
+  final String emojiName;
+  final bool selected;
+
+  @override
+  Widget build(BuildContext context) {
+    final parsed = tryParseEmojiCodeToUnicode(emojiCode);
+    if (parsed == null) {
+      return _TextEmoji(emojiName: emojiName, selected: selected);
+    }
+    final textScaler = MediaQuery.textScalerOf(context);
+    return SizedBox(
+      width: textScaler.scale(17),
+      child: Text(
+        style: const TextStyle(fontSize: 17),
+        strutStyle: const StrutStyle(fontSize: 17, forceStrutHeight: true),
+        parsed));
```

Comment:
This can partly be seen as a downstream consequence of not bundling an emoji font.

Maybe the right approach to keep things moving is:
* Have a `defaultTargetPlatform` check that makes a couple of things different here between iOS and Android.
* On iOS, find whatever value goes instead of 14.5 that gets a size of 17 with Apple's emoji font.
* On Android, use that value of 14.5 that gets a size of 17 with the Noto emoji font.
* That will make it work exactly as desired on iOS devices and Pixel devices. It might look a little odd on other Android devices — not sure how much.
* And then as a followup issue, we'll go back to bundling Noto Color Emoji, but only on non-Apple platforms because Apple platforms aren't compatible with it. Then this will work exactly as desired on all devices.

Reply:
Ah. Thanks for that investigation and proposed way forward!

Reply:
Things are different on iOS—a font size of 17 gives emoji that look (to me) approximately 17px square…but they're uncentered, and the `Text` takes 20px x 21.3px, when not constrained by the `SizedBox`. And the emoji doesn't look centered in the space occupied by the `Text`.

I think this is https://github.com/flutter/flutter/issues/119623 and https://github.com/flutter/flutter/issues/28894. I'm going to try this workaround for it (which I see you've 👍-reacted before): https://github.com/flutter/flutter/issues/28894#issuecomment-1717042574

Reply:
> I'm going to try this workaround for it (which I see you've 👍-reacted before): […]

Actually I'm not very excited about doing this; I'll try something simpler.